public/scripts/chats.js

```diff
@@ -330,19 +341,41 @@ export function hasPendingFileAttachment() {
  * @param {File} file File object
  * @returns {Promise<void>}
  */
-async function onFileAttach(file) {
-    if (!file) return;
+async function onFileAttach(files) {
+    if (!files || files.length === 0) return;
 
-    const isValid = await validateFile(file);
+    let allValid = true;
+    let totalSize = 0;
+    let fileNames = [];
 
-    // If file is binary
-    if (!isValid) {
+    // Validate all files and collect information
+    for (let i = 0; i < files.length; i++) {
+        const file = files[i];
+        const isValid = await validateFile(file);
+
+        if (!isValid) {
+            allValid = false;
+            break;
+        }
+
+        totalSize += file.size;
+        fileNames.push(file.name);
+    }
+
+    // If any file is invalid
+    if (!allValid) {
         $('#file_form').trigger('reset');
         return;
     }
 
-    $('#file_form .file_name').text(file.name);
-    $('#file_form .file_size').text(humanFileSize(file.size));
+    // Update UI to show multiple files
+    if (files.length === 1) {
+        $('#file_form .file_name').text(fileNames[0]);
```

Comment:
This array is only used to populate a displayed file name, so it can be safely removed.
```suggestion
        $('#file_form .file_name').text(files[0].name);
```